### PR TITLE
feat(ironfish): Manage pending state for assets

### PR DESCRIPTION
**Pull request by rohanjadvani**
## Summary

* Adds `blockHash` and `sequence` to asset records
  * There is no accompanying migration since this store is currently empty, so no records need to be migrated
* Adjusts backfill migration to populate `blockHash` and `sequence`
* Connects mints and burns when pending transactions are added to an account
* Increments supply only if connected mints and burns if the associated transaction is on a block

## Testing Plan

Added and adjusted unit tests

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/wallet/account.ts [hunk 1]
```diff
@@ -259,19 +269,23 @@ export class Account {
         'Existing asset owner should match public address',
       )
 
-      const existingSupply = existingAsset.supply
-      const supply = existingSupply - value
+      let supply = existingAsset.supply
+      if (blockHash && sequence) {
```
Comment (hughy) [inline ironfish/src/wallet/account.ts:273]: If there's no blockHash and sequence we won't update anything on the asset. Should we return early?

File: ironfish/src/wallet/account.ts [hunk 2]
```diff
@@ -398,19 +416,19 @@ export class Account {
...
+        transaction,
+        blockHash: null,
+        sequence: null,
+        submittedSequence,
+        timestamp: new Date(),
+        assetBalanceDeltas,
+      }
+
+      await this.saveConnectedMintsToAssetsStore(transactionValue, tx)
+      await this.saveConnectedBurnsToAssetsStore(transactionValue, tx)
```
Comment (hughy) [inline ironfish/src/wallet/account.ts:429]: We may not need to process pending burns at all if supply only changes on connected burns

File: ironfish/src/wallet/account.ts [hunk 3]
```diff
@@ -398,19 +416,19 @@ export class Account {
...
+      const transactionValue = {
+        transaction,
+        blockHash: null,
+        sequence: null,
+        submittedSequence,
+        timestamp: new Date(),
+        assetBalanceDeltas,
+      }
+
+      await this.saveConnectedMintsToAssetsStore(transactionValue, tx)
```
Comment (hughy) [inline ironfish/src/wallet/account.ts:428]: nit - wondering if the `Connected` still makes sense in the name if we call this for pending transactions?

File: ironfish/src/wallet/account.ts [hunk 4]
```diff
@@ -186,29 +186,26 @@ export class Account {
...
+
+      await this.walletDb.saveTransaction(this, transaction.hash(), transactionValue, tx)
     })
 
     return assetBalanceDeltas
   }
 
   async saveConnectedMintsToAssetsStore(
-    transaction: Transaction,
+    { blockHash, sequence, transaction }: TransactionValue,
```
Comment (hughy) [inline ironfish/src/wallet/account.ts:208]: I didn't know you could do this in TypeScript 🤯

Comment (rohanjadvani) [reply ironfish/src/wallet/account.ts:208]: :cowboy_hat_face:

File: ironfish/src/wallet/account.ts (hunk 2 above)
Comment (rohanjadvani) [reply ironfish/src/wallet/account.ts]: Yep removed

**Review by hughy — APPROVED** [review]

**Pull request merged**